File: backend/app/schemas/create_dossier.py
from pydantic import BaseModel, Field


class EntreprisePayload(BaseModel):
    ice: str
    raisonSociale: str
    rc: str = ""
    secteur: str = ""
    documentNames: list[str] = Field(default_factory=list)


class FinancementPayload(BaseModel):
    nature: str
    montantDemande: float
    valeurBien: float
    dureeMois: int
    apport: float
    urgence: str


class FournisseurBienPayload(BaseModel):
    fournisseur: str
    proformaReference: str
    proformaFileName: str | None = None
    natureBien: str
    etat: str
    valeurHt: float
    valeurTtc: float


class CreateDossierPayload(BaseModel):
    entreprise: EntreprisePayload
    financement: FinancementPayload
    fournisseurBien: FournisseurBienPayload


class CreateDossierResponse(BaseModel):
    id: str
    status: str
    message: str
    job_id: str | None = None
    stream_url: str | None = None
    result_url: str | None = None
    synthese_url: str | None = None
    filename: str | None = None


class StoredFileMeta(BaseModel):
    name: str
    objectKey: str
    size: int
    contentType: str
    category: str


class StoredDossierRecord(BaseModel):
    id: str
    name: str
    sector: str
    amount: int
    duration: int
    score: int = 0
    status: str = "pending"
    analyst: str
    receivedDaysAgo: int = 0
    date: str
    urgency: str | None = None
    receivedLabel: str | None = None
    ice: str
    rc: str = ""
    nature: str
    valeurBien: float
    apport: float
    fournisseur: str
    proformaReference: str
    natureBien: str
    etat: str
    valeurHt: float
    valeurTtc: float
    files: list[StoredFileMeta] = Field(default_factory=list)
    decisionDate: str | None = None
    analyseJobId: str | None = None
    analyseStatus: str | None = None
    analyse: dict | None = None

File: frontend/src/types/create-dossier.ts
import type { Urgency } from '@/types/dossier'

export type FinancingNature = 'mobilier' | 'immobilier'
export type BienEtat = 'neuf' | 'occasion'


export interface UploadedFileMeta {
  id: string
  name: string
  size: number
  mimeType: string
  
  file?: File
}

export interface EntrepriseFormData {
  ice: string
  raisonSociale: string
  rc: string
  
  secteurPreset: string
  
  secteurAutre: string
  documents: UploadedFileMeta[]
}

export interface FinancementFormData {
  nature: FinancingNature | ''
  montantDemande: string
  valeurBien: string
  dureeMois: string
  apport: string
  urgence: Urgency | ''
}

export interface FournisseurBienFormData {
  fournisseur: string
  proformaReference: string
  proformaFile: UploadedFileMeta | null
  natureBien: string
  etat: BienEtat | ''
  valeurHt: string
  valeurTtc: string
}

export interface CreateDossierFormState {
  entreprise: EntrepriseFormData
  financement: FinancementFormData
  fournisseurBien: FournisseurBienFormData
}


export interface CreateDossierPayload {
  entreprise: {
    ice: string
    raisonSociale: string
    rc: string
    secteur: string
    documentNames: string[]
  }
  financement: {
    nature: FinancingNature
    montantDemande: number
    valeurBien: number
    dureeMois: number
    apport: number
    urgence: Urgency
  }
  fournisseurBien: {
    fournisseur: string
    proformaReference: string
    proformaFileName: string | null
    natureBien: string
    etat: BienEtat
    valeurHt: number
    valeurTtc: number
  }
}

export interface CreateDossierResponse {
  id: string
  status: string
  message: string
  job_id?: string | null
  stream_url?: string | null
  result_url?: string | null
  synthese_url?: string | null
  filename?: string | null
}

export const SECTEURS = [
  'Transport',
  'Immobilier',
  'BTP',
  'Santé',
  'Industrie',
  'Commerce',
  'Agriculture',
  'Tourisme',
  'Tech & Services',
  'Énergie',
  'Automobile',
  'Éducation',
  'Autre',
] as const
